src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1317,10 +1317,26 @@ class Simplex_tree {
    * already present, its filtration value is not modified, unlike with other insertion functions. */
   template <class VertexRange>
   void insert_batch_vertices(VertexRange const& vertices, Filtration_value filt = 0) {
+    // When Options::link_nodes_by_label, update_simplex_tree_after_node_insertion must be called only if
+    // not already inserted - copy vertex handles from flat_map
+    std::vector<Vertex_handle> old_members(root_.members().size());
+    if constexpr (Options::link_nodes_by_label) {
+      for (auto sh = root_.members().begin(); sh != root_.members().end(); sh++)
+        old_members.emplace_back(sh->first);
```

Comment:
You create old_members that contains n copies of 0 then append the rest to it? Did you mean to use reserve instead?